src/olympia/amo/tests/__init__.py

```diff
@@ -589,19 +589,25 @@ def change_channel_for_addon(self, addon, listed):
             version.update(channel=channel)
 
     @classmethod
-    def make_addon_promoted(cls, addon, group_id, approve_version=False, apps=None):
+    def make_addon_promoted(
```

Comment:
I think the main reason you need the `reset` argument is because of the large tests.. right? if they were smaller, you wouldn't need this. This abstraction is probably not worth it IMO.
- It doesn't save much as you can delete all promotions in a single line anyway and now you have to go back here to understand what it does.
- True by default means calls to `make_addon_promoted` implicitly delete all other promotions (very un-intuitive)

Reply:
> I think the main reason you need the reset argument is because of the large tests.. right? 

No, its because of the existing tests that use `make_addon_promoted()` and are designed only to expect one promotion at a time. So if I were to do this:

> It doesn't save much as you can delete all promotions in a single line anyway and now you have to go back here to understand what it does.

...I'd have to touch a lot more test files.

Reply:
I think I +1 to Kevin here - _most_ tests that use `make_addon_promoted` will only be setting a single promotion in that test, so we just need to update the handful that make multiple `make_addon_promoted` calls in a single test to delete in between.  Some of _those_ tests may actually benefit from being expanded to test where is now multiple groups anyway.